lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -165,37 +176,36 @@ def prepare_data(d, init):
 
     def apply_callback(data):
         """A callback to apply changes."""
-        orig_xlim = axes.get_xlim()
-        orig_ylim = axes.get_ylim()
+        orig_limits = {
+            name: getattr(axes, f"get_{name}lim")()
+            for name in axis_map
+        }
 
         general = data.pop(0)
         curves = data.pop(0) if has_curve else []
         mappables = data.pop(0) if has_sm else []
         if data:
             raise ValueError("Unexpected field")
 
-        # Set / General
-        (title, xmin, xmax, xlabel, xscale, ymin, ymax, ylabel, yscale,
-         generate_legend) = general
+        title = general.pop(0)
+        axes.set_title(title)
+        generate_legend = general.pop()
+
+        for i, (name, axis) in enumerate(axis_map.items()):
+            axis_min = general[4*i]
+            axis_max = general[4*i + 1]
+            axis_label = general[4*i + 2]
+            axis_scale = general[4*i + 3]
+            if axis.get_scale() != axis_scale:
+                getattr(axes, f"set_{name}scale")(axis_scale)
```

Comment:
This one is accessible directly, though it's private:
```suggestion
                axis._set_axes_scale(axis_scale)
```
Not sure if we allow that here.

Reply:
`set_x/yscale` propagates to shared Axes. I think we want that behavior, so the proposed change does not work.